### PR TITLE
[fixes #3423] checks if .rubocop is a file before parsing

### LINKED ISSUE #3423: Breaks when a .rubocop directory exists
Rubocop errors when there is a directory named `.rubocop`
--------

## Expected behavior
Use the `.rubocop.yml` config

## Actual behavior
```
Is a directory @ io_fillbuf - fd:7 .rubocop
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/options.rb:38:in `readlines'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/options.rb:38:in `args_from_file'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/options.rb:19:in `parse'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/cli.rb:24:in `run'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/bin/rubocop:14:in `block in <top (required)>'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/bin/rubocop:13:in `<top (required)>'
/Users/joejuzl/.rbenv/versions/2.2.3/bin/rubocop:22:in `load'
/Users/joejuzl/.rbenv/versions/2.2.3/bin/rubocop:22:in `<main>'
```

## Steps to reproduce the problem
Run rubocop in a directory with a sub-directory named `.rubocop`.

## RuboCop version

```
$ rubocop -V
0.42.0
```

### PR DESCRIPTION
**Pull request by joejuzl**
### Problem:

Rubocop errors when there is a directory named .rubocop

### Expected behavior

Use the .rubocop.yml config

### Actual behavior

```
Is a directory @ io_fillbuf - fd:7 .rubocop
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/options.rb:38:in `readlines'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/options.rb:38:in `args_from_file'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/options.rb:19:in `parse'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/lib/rubocop/cli.rb:24:in `run'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/bin/rubocop:14:in `block in <top (required)>'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/Users/joejuzl/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.42.0/bin/rubocop:13:in `<top (required)>'
/Users/joejuzl/.rbenv/versions/2.2.3/bin/rubocop:22:in `load'
/Users/joejuzl/.rbenv/versions/2.2.3/bin/rubocop:22:in `<main>'
```

## Solution:

Check that `.rubocop` is not a directory before parsing.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/options.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def parse(command_line_args)
     private
 
     def args_from_file
-      if File.exist?('.rubocop')
+      if File.file?('.rubocop')
```
Comment (mikegee) [inline lib/rubocop/options.rb]: This fails for symbolic links and named pipes. (I'm not sure if we care.)

An alternative is:

```ruby
def args_from_file
  IO.readlines('.rubocop').map(&:strip)
rescue
  []
end
```

Comment (mikegee) [inline lib/rubocop/options.rb]: Or, maybe use [`File.readable?`](https://ruby-doc.org/core-2.2.0/File.html#method-c-readable-3F) instead of `File.file?`.

Comment (joejuzl) [inline lib/rubocop/options.rb]: It does actually work for symbolic links. 
I find the `rescue` solution a bit less clear... plus it hides potential error messages!
Is it worth it for named pipes?
Maybe `File.exists?('.rubocop') && !File.directory?('.rubocop')`? or is that overkill?

Comment (bbatsov) [inline lib/rubocop/options.rb]: This looks reasonable and clear to me - combining both checks.

**Comment by bbatsov** [comment]
You should also add some test illustrating the problem you've fixed.

**Comment by bbatsov** [comment]
The build is broken.

**Comment by joejuzl** [comment]
Build fixed!

**Comment by bbatsov** [comment]
You should also add some unit test for this and rebase on top of the current `master`.

**Comment by joejuzl** [comment]
spec added, rebased on master, and commits squashed

**Pull request merged**